### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by daliwali**
So it turns out that this problem can be solved with a `for` loop and the modulus operator. It could be even more memory efficient by modifying the input array in place, but I'm lazy.

### CODE REVIEW

**Comment by yanns** [comment]
:+1:

**Comment by emmenko** [comment]
:+1:

File: src/coffee/mixins.coffee [hunk 1]
```diff
@@ -99,9 +99,12 @@ module.exports =
   ###
   batchList: (list, size) ->
     throw new Error 'List batch size is required' unless size
-    _batch = (tickList, acc = []) ->
-      return acc if _.isEmpty tickList
-      acc.push _.head tickList, size
-      tail = _.tail tickList, size
-      _batch tail, acc
-    _batch(list)
+    batch = []
```
Comment (hajoeichler) [inline src/coffee/mixins.coffee]: Couldn't we just remove the accumulator?

Comment (daliwali) [inline src/coffee/mixins.coffee]: Dunno, but here's a more functional solution:

```js
function batch (array, size) {
  return array.reduce((batch, value, index) => {
    if (index % size === 0) batch.push([])
    batch[batch.length - 1].push(value)
    return batch
  }, [])
}
```

Comment (emmenko) [inline src/coffee/mixins.coffee]: Let's go for it :+1:

Comment (hajoeichler) [inline src/coffee/mixins.coffee]: I think that's nicer.

**Pull request merged**